tests/test_install.py

```diff
@@ -122,6 +127,48 @@ def test_calls_func(self):
         func.assert_called_with(some_path)
 
 
+@pytest.mark.parametrize("args", [
+    (),
+    ("one", ),
+    ("one", "two", "three", ),
+])
+def test_ensure_write_takes_two_arguments(args):
+    with pytest.raises(TypeError):
+        install.ensure_write(*args)
+
+
+class ensure_write_TestCase(unittest.TestCase):
+    def setUp(self):
+        self.prefix = generate_random_path()
+        self.dist = "foobar-%s-0" % random.randint(100, 200)
+        self.meta = {
+            "name": self.dist,
+            "files": ["a", "b", "c"],
+        }
+
+    @contextmanager
+    def generate_mock_check(self, return_value=True):
+        with patch.object(install, 'can_open_all_files_in_prefix') as mocked:
+            mocked.return_value = return_value
+            with patch.object(install, "load_meta") as load_meta:
+                load_meta.return_value = self.meta
+                yield mocked
+
+    def test_dispatches_to_can_open_all_files(self):
+        with self.generate_mock_check() as can:
+            install.ensure_write(self.prefix, self.dist)
+        can.assert_called_with(self.prefix, self.meta['files'])
+
+    def test_returns_none_on_true(self):
+        with self.generate_mock_check():
+            actual = install.ensure_write(self.prefix, self.meta)
```

Comment:
@tswicegood I'm trying to understand how this test is working. `self.meta` is a [dictionary](https://github.com/conda/conda/pull/1278/files#diff-6e55dd91d786eaa1da3d8f5311315efcR144), but the second argument of `ensure_write` gets passed to [`is_linked`](https://github.com/conda/conda/pull/1278/files#diff-fb1ca55b90c5cb761d62db03e6e0d406R496), and I'm pretty sure should be a string. I tried making some changes to this functionality and I am not getting test failures like

```
prefix = '/some/path/to/file120', dist = {'files': ['a', 'b', 'c'], 'name': 'foobar-111-0'}

    def is_linked(prefix, dist):
        """
        Return the install meta-data for a linked package in a prefix, or None
        if the package is not linked in the prefix.
        """
>       meta_path = join(prefix, 'conda-meta', dist + '.json')
E       TypeError: unsupported operand type(s) for +: 'dict' and 'str'
```